File: src/contests/cg/puzzle/the-fastest.ts
// 🎮 CodinGame Puzzle - the-fastest
// https://www.codingame.com/training/medium/the-fastest

// Times are HH:MM:SS so lexicographic comparison equals chronological comparison.
const n: number = parseInt(readline())
let best = ""
for (let i = 0; i < n; i++) {
  const t = readline().trim()
  if (best === "" || t < best) best = t
}
console.log(best)

File: src/contests/cg/puzzle/java/the-fastest.java
// 🎮 CodinGame Puzzle - the-fastest
// https://www.codingame.com/training/medium/the-fastest

import java.io.*;

class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        // Times are HH:MM:SS so lexicographic comparison equals chronological comparison.
        int n = Integer.parseInt(br.readLine().trim());
        String best = "";
        for (int i = 0; i < n; i++) {
            String t = br.readLine().trim();
            if (best.equals("") || t.compareTo(best) < 0) best = t;
        }
        System.out.println(best);
    }
}
